Realm/ObjectStore/object_store.cpp

```diff
@@ -400,9 +427,10 @@ void ObjectStore::update_realm_with_schema(Group *group, Schema const& old_schem
 
     // create tables
     create_metadata_tables(group);
-    create_tables(group, schema, migrating);
+    auto to_delete = create_tables(group, schema, migrating);
 
     if (!migrating) {
+        remove_properties(group, schema, to_delete);
```

Comment:
It doesn't make any sense to be calling remove_properties() in the not migrating case.